### PR TITLE
Implement pretty printing and proper stringification for records, constructors and lambdas.

### PR DESCRIPTION
**Pull request by RubenVerg**

### CODE REVIEW

File: shared/src/vyxal/VAny.scala [hunk 1]
```diff
@@ -134,11 +139,21 @@ end extension
...
+  override def toString = s"$name constructor"
 
 case class VObject(
     className: String,
     fields: Map[String, (Visibility, VAny)],
-)
+):
+  override def toString =
+    val fs = fields.map {
+      case (name, (vis, value)) => s"$value ${vis match
```
Comment (ysthakur) [inline shared/src/vyxal/VAny.scala]: I'd kinda prefer to have the match outside the string, assigned to a variable `prefix` or something

**Review by lyxal — CHANGES_REQUESTED** [review]
This only impacts when `toString` is called - printing a record doesn't call to string. This'll need to be changed.

**Comment by RubenVerg** [comment]
@lyxal fixed, thanks for spotting that

**Comment by lyxal** [comment]
For records, maybe something like:

```
<Name> { 
  !Public: value,
  $Restricted: value,
  =Private: value
}
```

For example,

```
<Map> {
  $keys: ["k1", "k2", "k3"],
  $values: ["v1", "v2", "v3"]
}
```

**Comment by RubenVerg** [comment]
@lyxal with the newlines and indentation? that's gonna become an issue when there are maps nested inside maps, it's gonna look wrong

**Comment by lyxal** [comment]
How about outermost map has indentation, and inner maps are flat

**Comment by ysthakur** [comment]
I feel like we should either go all the way with flatness or all the way with indentation. Maybe `toString` could be completely flat, but when displaying in the REPL, we pretty-print everything (including lists?) with indentation? And in the web interpreter, it could show objects like the web console shows JS objects, with dropdowns and stuff

**Review by lyxal — APPROVED** [review]

File: shared/src/vyxal/AST.scala [hunk 2]
```diff
@@ -184,6 +190,12 @@ enum Visibility derives CanEqual:
   case Private
   case Restricted
 
+  def sigil: String =
+    this match
```
Comment (ysthakur) [inline shared/src/vyxal/AST.scala]: You could also make this a field by adding it to visibility,s constru tor

Comment (ysthakur) [reply shared/src/vyxal/AST.scala]:
```scala
enum Visibility(val sigil: String) derives CanEqual:
  case Public extends Visibility("!")
  case Private extends Visibility("=")
  case Restricted extends Visibility("$")
```

**Review by ysthakur — APPROVED** [review]
LGTM, this should make working in the REPL more pleasant. I was going to suggest changing the interpolated strings to not be multiline by extracting the expressions into variables, but tbh, it doesn't matter much

**Pull request merged**